<launch>
    <!-- 🟢 RTSP 相機 node (取消註解使用) -->
  <node name="multi_rtsp_to_compressed_node"
        pkg="image_processing"
        type="rtsp_to_compressed_image_test.py"
        output="screen" />

  <node pkg="image_processing" type="fisheye_undistorter.py" name="undistorter1" output="screen">
      <param name="input_topic" value="/camera1/color/image_raw/compressed"/>
      <param name="output_topic" value="/camera1_fix/color/image_raw/compressed"/>
      <param name="factor" value="4.05"/>
      <param name="K" value="[[268.06554187, 1.51263281, 320],[0,356.57309093,231.71146684],[0,0,1]]"/>
      <param name="D" value="[[-0.15439177], [ 0.45612835], [-0.79521684], [ 0.46727377]]"/>
      <param name="balance" value="1.0"/>
      <param name="crop_fov_deg" value="90.0"/>
      <param name="full_hfov" value="150.0"/>
  </node>

  <!-- 🟢 Camera raw node (取消註解使用) -->
  <!-- <node pkg="image_processing" type="fisheye_undistorter.py" name="undistorter1" output="screen">
      <param name="input_topic" value="/camera1/color/image_raw/raw"/>
      <param name="output_topic" value="/camera1_fix/color/image_raw/raw"/>
      <param name="factor" value="4.05"/>
      <param name="K" value="[[268.06554187, 1.51263281, 320],[0,356.57309093,231.71146684],[0,0,1]]"/>
      <param name="D" value="[[-0.15439177], [ 0.45612835], [-0.79521684], [ 0.46727377]]"/>
      <param name="balance" value="1.0"/>
      <param name="crop_fov_deg" value="90.0"/>
      <param name="full_hfov" value="150.0"/>
  </node> -->

</launch>

